kubeflow/trainer/rhai/speculator_test.py

```
@@ -0,0 +1,1582 @@
+    trainer = SpeculativeDecodingTrainer(
+        verifier_model="meta-llama/Llama-3.1-8B-Instruct",
+        mode=SpeculatorMode.DATA_ONLY,
+        dataset_name="sharegpt",
+        output_dir="pvc://shared/datagen_output",
+    )
+
+    script = _render_speculator_training_script(trainer)
+
+    assert "_DATAGEN_SCRIPT_B64" in script
+    assert "base64.b64decode(_DATAGEN_SCRIPT_B64)" in script
+    assert "/tmp/data_generation_offline.py" in script
```

Ruff (0.15.21)

[error] Probable insecure usage of temporary file or directory: "/tmp/data_generation_offline.py"

(S108)